pkgs/by-name/ld/ld64/patches/0015-Set-the-version-string-in-the-build.patch

```diff
@@ -0,0 +1,41 @@
+From 40a0b0a41c14d580488477c584010558334dedf1 Mon Sep 17 00:00:00 2001
+From: Randy Eckenrode <randy@largeandhighquality.com>
+Date: Sun, 17 Nov 2024 10:02:34 -0500
+Subject: [PATCH 15/20] Set the version string in the build
+
+---
+ src/ld/meson.build | 11 +++++++++++
+ 1 file changed, 11 insertions(+)
+
+diff --git a/src/ld/meson.build b/src/ld/meson.build
+index 201e891..769ac28 100644
+--- a/src/ld/meson.build
++++ b/src/ld/meson.build
+@@ -11,6 +11,16 @@ InputFiles_cpp = configure_file(
+     output : '@PLAINNAME@',
+ )
+ 
++version_string_cpp = configure_file(
++    capture : true,
++    command : [
++        find_program('bash', native : true),
++        '-c',
++        'echo \'extern const char ld_classicVersionString[] = "@(#)PROGRAM:ld PROJECT:ld64-' + meson.project_version() + '\\n";\''
```

Comment:
Since we’re making some non‐trivial modifications, do we maybe want to include some sort of indicator of that in the version string – a `-nixpkgs` suffix or something?

Reply:
Most of the major changes were adapted from the previous patch set. Behavior should be unchanged. There’s also a risk that if we report something unusual that it will confuse or break linker version detection.